### PR TITLE
Feature/registration form 2

### PR DESCRIPTION
**Pull request by SwiftXLight**
**Description changes:** 

- add specialization, time zone, address, country, city, date of birth, gender columns.
- and validation for them.
- add auth guard for ":id" endpoint.

[link to Jira ticket](https://smartdc23.atlassian.net/browse/SMAR-33?atlOrigin=eyJpIjoiMmQxOGMxOTNiYjQxNDY0NTkyZTJiOThmNGI1NjVmYjEiLCJwIjoiaiJ9)

**PR checklist:**
1.1 Common
1.1.1 types for input and output params
1.1.2 no any, should be also added to eslint rules
1.1.3 try/catch for any async function
1.1.4 no magic numbers
1.1.5 compare only with constants not with strings
(instead of user === ‘admin’, better user === roles.admin)
1.1.6 no ternary operator inside ternary operator (bad example: question ? first:
second ? third: fourth)
1.1.7 avoid similar types with duplicating by generics
1.1.8 no console.log in pr
1.1.9 .env file should be in .gitignore
1.1.10 delete commented code if it's not part of documentation
1.1.11 no links in the code, env links should be in env file (for example: server url),
constant links (for example default avatar URL) should be in constant file
1.1.12 constants and function names should be lowercase.
1.1.13 no dates format in the code (like "dd/MM/yyyy”), move it in global constant
variable
1.1.14 import rules. imports should come in a specific order: node modules, absolute
path, relative path
1.1.15 strings should be in the constant variable. Example: instead of ‘15 3 * * *’,
const EACH_DAY_15h_15min = ‘15 3 * * *’

1.2 Backend
1.2.1 swagger for each endpoint
1.2.2 less requests to db, all data should be taken with one query
1.2.3 `public` in method use only is use function externally
1.2.4 use ConfigService instead of `process.env`

**Doctor table with new columns:**
image
**Try reach endpoint without access:**
image
**Doctor dto:**
image
**Doctor entity:**
image

### CODE REVIEW

File: src/modules/doctor/doctor.controller.ts [hunk 1]
```diff
@@ -24,14 +34,21 @@ export default class DoctorController {
...
-  @Get('/:id')
+  @Get('/id/:id')
   getOne(@Param('id') id: number): Promise<Doctor> {
     return this.doctorService.getDoctorByID(id);
   }
 
+  @ApiOperation({ summary: 'Getting doctor by email' })
+  @ApiResponse({ status: 200, type: Doctor })
+  @Get('/email/:email')
+  findByEmail(@Param('email') email: string): Promise<Doctor> {
```
Comment (lyudad) [inline src/modules/doctor/doctor.controller.ts]: hmmm when we need it?

File: src/modules/doctor/doctor.controller.ts [hunk 2]
```diff
@@ -24,14 +34,21 @@ export default class DoctorController {
...
+    return this.doctorService.getDoctorByEmail(email);
+  }
+
   @ApiOperation({ summary: 'Delete doctor by id' })
   @ApiResponse({ status: 200, type: [Doctor] })
-  @Delete('/:id')
+  @Delete('/id/:id')
```
Comment (lyudad) [inline src/modules/doctor/doctor.controller.ts]: route is not protected

File: src/modules/doctor/doctor.service.ts [hunk 3]
```diff
@@ -59,6 +59,19 @@ export default class DoctorService {
     }
   }
 
+  async getDoctorByEmail(email: string): Promise<Doctor | null> {
+    try {
+      const doctor = await this.doctorRepository
```
Comment (lyudad) [inline src/modules/doctor/doctor.service.ts]: no we use query builder only

File: src/modules/doctor/doctor.controller.ts (hunk 1 above)
Comment (SwiftXLight) [reply src/modules/doctor/doctor.controller.ts]: deleted

File: src/modules/doctor/doctor.controller.ts (hunk 2 above)
Comment (SwiftXLight) [reply src/modules/doctor/doctor.controller.ts]: add auth guard

File: src/modules/doctor/doctor.service.ts (hunk 3 above)
Comment (SwiftXLight) [reply src/modules/doctor/doctor.service.ts]: deleted (the same issue you write in first comment)

**Pull request closed without merging**